### ISSUE TITLE
Unable to play Overcooked 2 online with Pcap2socks

### ISSUE THREAD

**Issue by terryops** [zhxie/pcap2socks#21]
I'm testing Pcap2Socks and Surge Mac Router mode with the same socks5 proxy, the result is that Pcaps2socks failed to get inside the online game while Surge Mac doesn't. It just stucks at the room loading screen. 
I've pasted the log with -vv argument, and I also pasted the game's connection urls captured Surge, please check out the attachments below.
Helping me solving this problem is much appreciated.
[pcap2socks.log](https://github.com/zhxie/pcap2socks/files/7553587/pcap2socks.log)
[surge.csv](https://github.com/zhxie/pcap2socks/files/7553588/surge.csv)

**Comment by zhxie** [comment]
I found that maybe some connections were closed abnormally due to the reason of time out like `13.225.93.44`. It seems there is a bug in the TCP stack which will queue received data from proxy instead of sending back to the source. I will try to take a look at it, but I am a little short of time, I may not be able to fix the issue in time.

**Comment by zhxie** [comment]
Issue confirmed. I have tested Overcooked 2 in Nintendo Switch and found that the error may occurred in an online room.

**Comment by zhxie** [comment]
I presume that the problem is on L9067. The TCP stack used by pcap2socks is immature, duplicates in a short time will not trigger retransmission. And due to some reason, pcap2socks does not retransmit timed out packets to the source again. And finally, the connection get reset in L11708 due to time out.

**Comment by terryops** [comment]
Great! Will you be able to test it in a few days?
By the way, do you have donation option for your project.
 I’d like to donate since pcap2socks is running well I want to support it.

**Comment by zhxie** [comment]
Yes, I need some time to review codes.
 
No, pcap2socks does not accpet any forms of sponsorship or donation.
Also, pcap2socks is under [MIT license](https://github.com/zhxie/pcap2socks/blob/master/LICENSE). Any use, modification, distribution of the code is permitted but the author or copyright holders will never take liability or warranty.

**Comment by zhxie** [comment]
Would you like to try to run [this build](https://github.com/zhxie/pcap2socks/actions/runs/1494705196) with `-vv` argument attached again? I may need some extra information to debug.

By the way, which arch of Mac are you using, an Intel (x86) one or an Apple Silicon (aarch64) one?

**Comment by terryops** [comment]
Still no luck with the new build. 
I have both x86 and ARM Mac, the attachment is logged by a x86 one. If you need logs for ARM Mac, I can also make it for you, but it may not behave differently.
[pcap2socks.log](https://github.com/zhxie/pcap2socks/files/7592374/pcap2socks.log)

**Comment by terryops** [comment]
Any luck on fixing the bug?

**Comment by zhxie** [comment]
I suppose it is a bug in UDP, since there is no response packets back from the proxy. But I am short of time last weekend, I have not captured the packets and gone through the traffic yet.

**Comment by terryops** [comment]
Any progress?

**Comment by terryops** [comment]
Btw, could another project of yours(ikago) solve this problem?

**Comment by zhxie** [comment]
No, I have not maintained the code for a long time, and exactly I cannot find the cause of the bug. The issue is marked as help wanted, and someone may fix it.

I am not sure if Ikago works.
